packages/cf-component-box/src/Box.js

```diff
@@ -0,0 +1,141 @@
+import React, { PropTypes } from 'react';
+import { createComponent } from 'cf-style-container';
+import propertiesSpec from './propertiesSpec';
+
+const propertyNames = Object.keys(propertiesSpec);
+
+// Generates styles given the provided props.
+//
+// This is driven by the metadata provided in propertiesSpec
+// and produces a datastructure like this:
+//
+// {
+//   margin: props.margin || props.theme['margin']
+//   width: props.width || props.theme['width']
+//   ...
+// }
+//
+// See propertySpec.js for more detail
+//
+const styles = props => {
+  let styles = {};
+
+  const propertyNames = Object.keys(propertiesSpec);
+  propertyNames.forEach(name => {
```

Comment:
I don't like the for loop either. This is the only place that bothers me, since this is the only one that will get hit in a production build. With the current functionality, I can't think of a good way to avoid it. 

If we're ok with dropping the behavior of providing a default if a prop isn't passed to the component, the loop can be much better by looping over the prop names and comparing those to all the properties. In most cases the number of props will be small (<10) and we can exit early with no work by checking membership in propertiesSpec.

The other option is to unroll it and write a line per specced property, which is annoying from a dev perspective.